src/VideoCommand.cc

```diff
@@ -350,3 +360,194 @@ Json::Value FindVideo::construct_responses(
+
+    query.QueryNode(
+            get_value<int>(cmd, "_ref", -1),
+            VDMS_VID_TAG,
+            cmd["link"],
+            cmd["constraints"],
+            results,
+            get_value<bool>(cmd, "unique", false)
+            );
+
+    return 0;
+}
+
+Json::Value FindFrames::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    const Json::Value& cmd = json[_cmd_name];
+
+    Json::Value ret = check_responses(responses);
+
+    auto error = [&](Json::Value& res)
+    {
+        ret[_cmd_name] = res;
+        return ret;
+    };
+
+    Json::Value& FindFrames = responses[0];
+
+    bool flag_empty = true;
+
+    for (auto& ent : FindFrames["entities"]) {
+
+        if(!ent.isMember(VDMS_VID_PATH_PROP)){
+            continue;
+        }
+
+        std::string video_path = ent[VDMS_VID_PATH_PROP].asString();
+        ent.removeMember(VDMS_VID_PATH_PROP);
+
+        if (ent.getMemberNames().size() > 0) {
+            flag_empty = false;
+        }
+
+        try {
+            std::vector<unsigned> frames;
+
+            // Copy of operations is needed, as we pass the operations to
+            // the enqueue_operations() method of ImageCommands class, and
+            // it should not include 'interval' operation.
+            Json::Value operations  = cmd["operations"];
+
+            Json::ArrayIndex interval_idx;
+            bool is_interval = get_interval_index(cmd, interval_idx);
+            bool is_frames   = cmd.isMember("frames");
+
+            if (is_frames) {
+                for (auto& fr : cmd["frames"]) {
+                    frames.push_back(fr.asUInt());
+                }
+            }
+            else if (is_interval) {
+                Json::Value interval_op = operations[interval_idx];
+                int i;
+                for (
+                    i =  get_value<int>(interval_op, "start");
+                    i <  get_value<int>(interval_op, "stop");
+                    i += get_value<int>(interval_op, "step"))
+                {
+                    frames.push_back(i);
+                }
+                Json::Value deleted;
+                operations.removeIndex(interval_idx, &deleted);
+            }
+            else {
+                // This should never happen, as we check this condition in
+                // FindFrames::construct_protobuf(). In case this happens, it
+                // is better to signal it rather than to continue
+                Json::Value return_error;
+                return_error["status"]  = RSCommand::Error;
+                return_error["info"] = "No 'frames' or 'interval' parameter";
+                return error(return_error);
+            }
+
+            VCL::Video video(video_path);
+            FindImage  image_op;
+            VCL::Image::Format format = image_op.get_requested_format(cmd);
+
+            if (format == VCL::Image::Format::NONE_IMAGE) {
```

Comment:
I would call it NO_IMAGE or IMAGE_NONE.

Reply:
good point, but for a separate PR (this comes from the image class, which is not touched here).